### DISCUSSION TITLE
Understanding Token Swaps and Contract Limitations on PancakeSwap

### DISCUSSION THREAD

**Discussion by talentfreelancer** [community/community#143471] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How can I swap tokens and receive the swapped tokens directly in their respective token contracts? As you may know, PancakeSwap does not permit transfers to token0 and token1 contracts.

**Comment by personalcodecare** [comment]
address[] memory path = new address;
path[0] = address(this);
path[1] = pairToken;
path[2] = router.WETH();

    _approve(address(this), address(router), amount_);
    try router.swapExactTokensForETHSupportingFeeOnTransferTokens(
        amount_, 
        0, 
        path, 
        address(this), 
        block.timestamp
    ){}
    catch Error(string memory reason) {
        console.log(reason);
    }

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/143474#discussioncomment-11145455) for reference to why this discussion was closed and locked. Thanks!
